compiler/plugins/target/AMD-AIE/iree-amd-aie/IR/AMDAIEOps.td

```diff
@@ -166,6 +166,66 @@ def AMDAIE_WorkgroupOp : AMDAIE_Op<"workgroup",
   let hasVerifier = 1;
 }
 
+//===----------------------------------------------------------------------===//
+// IREE AMDAIE Buffer And Lock Ops
+//===----------------------------------------------------------------------===//
+
+def AMDAIE_BufferOp: AMDAIE_Op<"buffer", [
+    DeclareOpInterfaceMethods<OpAsmOpInterface, ["getAsmResultNames"]>
+  ]>, Results<(outs AnyMemRef)> {
+  let summary = "Represents a buffer on an AIE tile.";
+  let description = [{
+    This operation represents a buffer on an AIE tile. The buffer can have an
+    optional address, indicating the location of the buffer on the tile.
+    
+    Example:
+
+    ```mlir
+    %tile = amdaie.tile(%c0, %c0)
+    %buffer = amdaie.buffer(%tile, 1024)
+    ```
+  }];
+
+  let arguments = (
+    ins Index:$tile,
+        OptionalAttr<UI32Attr>:$address
+  );
+
+  let results = (outs AnyMemRef:$buffer);
+
+  let assemblyFormat = [{ 
+    `(` $tile (`,` $address^)? `)` attr-dict `:` type($buffer)
+  }];
+}
+
+def AMDAIE_LockOp: AMDAIE_Op<"lock", [
+    Pure,
+    DeclareOpInterfaceMethods<OpAsmOpInterface, ["getAsmResultNames"]>
+  ]>, Results<(outs Index)> {
+  let summary = "Represents a physical lock on an AIE tile.";
+  let description = [{
+    This operation represents a lock on an AIE tile. The operation is fully 
+    specified through a tile and lock ID value, designating the exact physical 
+    lock to be used. This op helps with guaranteeing/verifying correct reuse of
+    the same lock. The op accepts an optional initialization value.
+
+    Example:
+
+    ```mlir
+    %tile = amdaie.tile(%c0, %c1)
+    %lock = amdaie.lock(%tile(0), 2)
```

Comment:
What does `%tile(0)` and `2` represent? It's good to add more descriptions.

Reply:
Added it to the description.